scarlet/display.py

```diff
@@ -1,224 +1,268 @@
-        """Map the `value` onto [0,1]
+        Parameters
+        ----------
+        img: array_like
+            Image to normalize
+        percentile: array_like, default=[1,99]
+            Lower and upper percentile to consider. Pixel values below will be
+            set to zero, above to saturated.
         """
-        if self.vmin is None:
-            vmin = np.min(value)
-        else:
-            vmin = self.vmin
-        if self.vmax is None:
-            vmax = np.max(value)
-        else:
-            vmax = self.vmax
-        self._set_scale(vmin, vmax)
-        result = self.asinh(value)
-        xmin, xmax = self.get_range()
-        result = (result-xmin)/(xmax-xmin)
-        return result
-
-
-class Linear(Normalize):
-    """Use a linear interpolation to map image intensities in matplotlib
-
-    This class is used in order to map the intensities to the range [0,1].
-    """
-    def __init__(self, vmin=None, vmax=None, img=None, clip=False):
-        """Initialize
+        assert len(percentiles) == 2
+        vmin, vmax = np.percentile(img, percentiles)
+        super().__init__(vmin=vmin, vmax=vmax)
+
+class AsinhNorm(Norm):
+    def __init__(self, vmin=0, beta=1):
+        """Asinh norm
+
+        Arcsinh scaling, see Lupton et al., 2004, PASP 116, 133–137
 
         Parameters
         ----------
-        vmin: float, default=`None`
+        vmin: float, default 0
             Minimum pixel value.
-            If `vmin` is `None`, the minimum value of `img` will be used.
-        vmax: float, default=`None`
-            Maximum pixel value.
-            If `vmin` is `None`, the maximum value of `img` will be used.
-        img: array_like, default=`None`
-            This should be an array with dimensions (bands, height, width).
-            Either `vmin` and `vmax` or `img` is required to create an Asinh mapping
-        clip: bool, default=`False`
-            Whether or not to clip values.
-            This is a default from matplotlib, but clip=True is currently not supported.
+        beta: float, default=1
+            Softening parameter
         """
-        Normalize.__init__(self, vmin, vmax, clip)
-        if img is not None:
-            vmin, vmax = self._get_scale(vmin, vmax, img)
-        if vmin is not None and vmax is not None:
-            self._set_scale(vmin, vmax)
-
-    def _get_scale(self, vmin, vmax, img):
-        if vmin is None:
-            vmin = np.ma.min(img)
-        if vmax is None:
-            vmax = np.ma.max(img)
-        return vmin, vmax
-
-    def _set_scale(self, vmin, vmax):
         self.vmin = vmin
-        self.vmax = vmax
-        self.data_range = vmax-vmin
+        self.beta = beta
+
+    def __call__(self, img):
+        """Apply the norm to RGB channels
+
+        Parameters
+        ----------
+        img: array_like
+            Image to normalize
+
+        Returns
+        -------
+        normalized img
+        """
+        img -= self.vmin
```

Comment:
That's better, yes. It was OK since this method is general, but it is *meant* to be run on the RGB channel values, not the original images. See `img_to_rgb`.